app/workers/service_discovery/import_cluster_service_definitions_worker.rb

```diff
@@ -4,25 +4,28 @@ module ServiceDiscovery
   class ImportClusterServiceDefinitionsWorker
     include Sidekiq::Worker
 
-    def perform(account_id, cluster_namespace, service_name)
-      return unless ThreeScale.config.service_discovery.enabled
-
-      account = Account.providers.find account_id
-      service_system_name = [cluster_namespace, service_name].join('-')
+    def self.create_service(account, cluster_namespace, cluster_service_name)
+      perform_async(:create, account_id: account.id, cluster_namespace: cluster_namespace, cluster_service_name: cluster_service_name)
+    end
 
-      cluster = ServiceDiscovery::ClusterClient.new
-      cluster_service = cluster.find_discoverable_service_by(name: service_name,
-                                                             namespace: cluster_namespace)
+    def self.resync_service(service)
+      perform_async(:resync, service_id: service.id)
+    end
 
-      service_creation = ServiceCreationService.call(account, name: service_name,
-                                                              system_name: service_system_name,
-                                                              kubernetes_service_link: cluster_service.self_link)
-      new_api = service_creation.service
+    def perform(action, options = {})
+      return unless ThreeScale.config.service_discovery.enabled
 
-      return unless service_creation.success? && new_api.persisted?
+      opts = options.dup.symbolize_keys
+      import_definitions_service = ImportClusterServiceDefinitionsService.new
 
-      new_api.import_cluster_service_endpoint(cluster_service)
-      new_api.import_cluster_active_docs(cluster_service)
+      case action.to_sym
```

Comment:
I'll refactor this as it violates Single Responsibility. Better having 2 separate workers, I think.